### DISCUSSION TITLE
postgres server_default as PK

### DISCUSSION THREAD

**Discussion by cmflynn** [collerek/ormar#757] [Q&A]
Hey guys, perhaps I'm missing something basic but when I tried to use a postgres server_default func as a primary key, i get errors on model.save().  here is the issue I opened up: https://github.com/collerek/ormar/issues/744. any help is appreciated. ty 🙏

**Comment by SepehrBazyar** [comment]
Hi @cmflynn 
I am not very sure, but maybe [this part](https://fastapi-utils.davidmontague.xyz/user-guide/basics/guid-type/#server-default) of the `fastapi-utils` doucment related to `server_default` settings is the answer:
> Note this will only work if you have installed the pgcrypto extension in your postgres instance. If the user you connect with has the right privileges, this can be done by calling the fastapi_utils.guid_type.setup_guids_postgresql function.

The settings required are very simple just run this:
`engine.execute('create EXTENSION if not EXISTS "pgcrypto"')`

**Reply by cmflynn** [reply]
interesting -- will look into this. tyvm

**Reply by SepehrBazyar** [reply]
> interesting -- will look into this. tyvm

I would be happy to know if this was the reason or not.
